Event/ElasticsearchDocumentEvent.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the ONGR package.
+ *
+ * (c) NFQ Technologies UAB <info@nfq.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchBundle\Event;
+
+use ONGR\ElasticsearchBundle\Client\Connection;
+use ONGR\ElasticsearchBundle\Document\DocumentInterface;
+
+/**
+ * Event to be dispatched in various Elasticsearch methods, containing associated document.
+ */
+class ElasticsearchDocumentEvent extends ElasticsearchEvent
```

Comment:
Create `AbstractElasticsearchEvent` with connection holder. Then create ElasticsearchCommitEvent and ElasticsearchPersistEvent classes where those will extend abstract class.